pkg/models/blackout_dates_test.go

```
@@ -64,3 +61,54 @@ func (suite *ModelSuite) Test_FetchTSPBlackoutDates() {
 		t.Errorf("Blackout dates query should have returned one result but returned zero instead.")
 	}
 }
+
+// Need a test where the GBLOC does and doesn't match
+func (suite *ModelSuite) Test_FetchTSPBlackoutDatesWithGBLOC() {
+	t := suite.T()
+	// Use FetchTSPBlackoutDates on two queries: one that should use a market value and one that doesn't.
+	// Create one blackout date object with a market.
+	tsp, _ := testdatagen.MakeTSP(suite.db, "A Very Excellent TSP", "XYZA")
+	tdl, _ := testdatagen.MakeTDL(suite.db, "Oklahoma", "62240", "5")
+	blackoutStartDate := time.Now()
+	blackoutEndDate := blackoutStartDate.Add(time.Hour * 24 * 2)
+	pickupDate := blackoutStartDate.Add(time.Hour)
+	market1 := "dHHG"
+	sourceGBLOC1 := "OHAI"
+	sourceGBLOC2 := "OHNO"
+	testdatagen.MakeBlackoutDate(suite.db, tsp, blackoutStartDate, blackoutEndDate, &tdl, &sourceGBLOC1, &market1)
+
+	// Create two ShipmentWithOffers, one using the first set of times and a market, the other using the same times but without a market.
+	shipmentWithOfferInGBLOC1 := models.ShipmentWithOffer{
+		ID: uuid.Must(uuid.NewV4()),
+		TrafficDistributionListID: tdl.ID,
+		PickupDate:                pickupDate,
+		SourceGBLOC:               &sourceGBLOC1,
+		Market:                    &market1,
+		AdministrativeShipment:    swag.Bool(false),
+		BookDate:                  testdatagen.DateInsidePerformancePeriod,
+	}
+
+	shipmentWithOfferInGBLOC2 := models.ShipmentWithOffer{
+		ID: uuid.Must(uuid.NewV4()),
+		TrafficDistributionListID: tdl.ID,
+		PickupDate:                pickupDate,
+		SourceGBLOC:               &sourceGBLOC2,
+		Market:                    nil,
+		AdministrativeShipment:    swag.Bool(false),
+		BookDate:                  testdatagen.DateInsidePerformancePeriod,
+	}
+
+	fetchWithMatchingGBLOC, err := FetchTSPBlackoutDates(suite.db, tsp.ID, shipmentWithOfferInGBLOC1)
+	if err != nil {
+		t.Errorf("Error fetching blackout dates.")
+	} else if len(fetchWithMatchingGBLOC) == 0 {
```

I think these tests read better when they're in the negative: "If this isn't the case, we know we have a problem." Plus, it's more specific. What if `len() == 3`? I have no idea how that'd happen, but we'd want to throw an error.